projects/packages/connection/src/class-rest-connector.php

```diff
@@ -947,6 +986,40 @@ public static function disconnect_site() {
 		);
 	}
 
+	/**
+	 * Unlinks current user from the WordPress.com Servers.
+	 *
+	 * @since 6.3.3
+	 *
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 *
+	 * @return bool|WP_Error True if user successfully unlinked.
+	 */
+	public static function unlink_user( $request ) {
+
+		if ( ! isset( $request['linked'] ) || false !== $request['linked'] ) {
+			return new WP_Error( 'invalid_param', esc_html__( 'Invalid Parameter', 'jetpack-connection' ), array( 'status' => 404 ) );
+		}
+
+		// Allow admins to force a disconnect by passing the "force" parameter
+		// This allows an admin to disconnect themselves
+		if ( isset( $request['force'] ) && false !== $request['force'] && current_user_can( 'manage_options' ) && ( new Manager( 'jetpack' ) )->disconnect_user_force( get_current_user_id() ) ) {
```

Comment:
I'm finding it a bit weird to have the actual disconnect function be invoked in the if statement checks. I guess it works correctly, and I can see the benefit of a simpler hierarchy of if statements but I don't think it's best practices and, imo, it makes it harder to read and understand what's happening 🤔

Reply:
I disagree on this one - since the function returns a boolean, I think It's OK to run in as part of the `if` statement, given that the return value of the function is intended to be a rest_response object or WP_Error.

The first couple of checks in the if statement are used to bail early before trying the disconnect.

The second `else if` condition follows the same pattern to determine if the disconnect operation worked or failed.

Reply:
I don't think this is a big enough issue to push back too hard, so I won't block approval for it again 😄  However I just wanted to share an article I found that explains, in better words than I could, why I am on the other side of this debate: https://www.teamten.com/lawrence/programming/keep-if-clauses-side-effect-free.html